### PR TITLE
The great research hugboxing

### PR DESCRIPTION
**Pull request by iloveloopers**
# About the pull request
removes the ability to buy clearance papers
clearance cards now give credits instead
cards no longer lose their value when scanned by the wrong person

# Explain why it's good for the game
less opportunities for bald researchers to waste the team's hard earned credits

# Testing Photographs and Procedure
i tested it and it works

# Changelog
:cl:
del: Removed ability to buy clearance papers.
balance: Clearance cards now give their equivalent clearance in credits.
balance: Clearance cards no longer lose their value when scanned by the wrong person.
/:cl:

### CODE REVIEW

**Comment by TheManWithNoHands** [comment]
Dont remove buying chem papers. It can often come in clutch when IOs already piked the colonie clean.
Great source of legendary propertys.

Dont remove tools just because some dont know how to use it.

**Comment by iloveloopers** [comment]
Right, because rolling the dice on RNG clearance papers in hopes of getting a legendary is peak game design. 99% of the time they aren't bought at all and the 1% is bald researchers screwing up their team by wasting credits on it.

File: code/modules/reagents/chemical_research/Chemical-Research.dm [hunk 1]
```diff
@@ -3,6 +3,7 @@ GLOBAL_DATUM_INIT(chemical_data, /datum/chemical_data, new)
 /datum/chemical_data
 	var/rsc_credits = 0
 	var/clearance_level = 1
+	var/credits_gained = 0 //credits gained from survivor clearance cards
```
Comment (Drulikar) [inline code/modules/reagents/chemical_research/Chemical-Research.dm:6]: Either autodoc it properly with `///` or name it better

File: code/game/machinery/computer/research.dm [hunk 2]
```diff
@@ -59,36 +59,23 @@
...
-	else
-		give_level = card.clearance_access
-
-	GLOB.chemical_data.clearance_level = give_level
-	if(give_x)
-		GLOB.chemical_data.clearance_x_access = TRUE
-		GLOB.chemical_data.reached_x_access = TRUE
+	var/credits_to_add = max(card.credits_to_give - GLOB.chemical_data.credits_gained, 0)
+	GLOB.chemical_data.update_credits(credits_to_add)
+	GLOB.chemical_data.credits_gained = max(GLOB.chemical_data.credits_gained, card.credits_to_give)
```
Comment (Drulikar) [inline code/game/machinery/computer/research.dm:75]:
```suggestion
	if(credits_to_add)
		GLOB.chemical_data.update_credits(credits_to_add)
		GLOB.chemical_data.credits_gained += credits_to_add
```
Existing way seems needlessly convoluted for what you're trying to accomplish of just limiting the bonus to the best card swiped.

File: code/game/machinery/computer/research.dm [hunk 3]
```diff
@@ -59,36 +59,23 @@
...
 
-	visible_message(SPAN_NOTICE("[user] swipes their ID card on \the [src], updating the clearance to level [give_level][give_x ? "X" : ""]."))
-	msg_admin_niche("[key_name(user)] has updated the research clearance to level [give_level][give_x ? "X" : ""].")
+	visible_message(SPAN_NOTICE("[user] swipes their ID card on \the [src], granting [credits_to_add] credits."))
+	msg_admin_niche("[key_name(user)] has swiped a clearance card and given [credits_to_add] credits to research.")
```
Comment (Drulikar) [inline code/game/machinery/computer/research.dm:78]:
```suggestion
	msg_admin_niche("[key_name(user)] has swiped a clearance card to give [credits_to_add] credits to research.")
```

File: code/game/machinery/computer/research.dm [hunk 4]
```diff
@@ -59,36 +59,23 @@
...
-	if(give_x)
-		GLOB.chemical_data.clearance_x_access = TRUE
-		GLOB.chemical_data.reached_x_access = TRUE
+	var/credits_to_add = max(card.credits_to_give - GLOB.chemical_data.credits_gained, 0)
+	GLOB.chemical_data.update_credits(credits_to_add)
+	GLOB.chemical_data.credits_gained = max(GLOB.chemical_data.credits_gained, card.credits_to_give)
 
-	visible_message(SPAN_NOTICE("[user] swipes their ID card on \the [src], updating the clearance to level [give_level][give_x ? "X" : ""]."))
-	msg_admin_niche("[key_name(user)] has updated the research clearance to level [give_level][give_x ? "X" : ""].")
+	visible_message(SPAN_NOTICE("[user] swipes their ID card on \the [src], granting [credits_to_add] credits."))
```
Comment (Drulikar) [inline code/game/machinery/computer/research.dm:77]:
```suggestion
	visible_message(SPAN_NOTICE("[user] swipes their ID card on [src], granting [credits_to_add] credits."))
```

**Review by Drulikar — CHANGES_REQUESTED** [review]

**Comment by iloveloopers** [comment]
its jover

**Pull request closed without merging**